src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/diagnostics/support/BuildMemoryComponent.java

```diff
@@ -0,0 +1,128 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.support;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.PrintedContent;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.Messages;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.BuildMemoryReport;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.diagnostics.Diagnostics;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.ToGerritRunListener;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.model.BuildMemory;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.GerritTriggeredEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.security.Permission;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Support Core Component writing the {@link BuildMemoryReport}.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+@Extension(optional = true)
+public class BuildMemoryComponent extends Component {
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Diagnostics.getRequiredPermission());
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Gerrit: " + Messages.BuildMemoryReport_DisplayName();
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        container.add(new PrintedContent("gerrit/buildmemory.md") {
+            @Override
+            protected void printTo(PrintWriter out) throws IOException {
+                ToGerritRunListener toGerritRunListener = ToGerritRunListener.getInstance();
+                if (toGerritRunListener == null) {
+                    //logging is performed by getInstance() above
+                    return;
+                }
+                SimpleDateFormat tsFormat = BuildMemoryReport.TS_FORMAT;
+                BuildMemoryReport report = toGerritRunListener.report();
+                out.println("#" + getDisplayName());
+                out.println();
+
+                List<Map.Entry<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>>> entries =
+                        report.getSortedEntrySet();
+
+                for (Map.Entry<GerritTriggeredEvent, List<BuildMemory.MemoryImprint.Entry>> entry : entries) {
```

Comment:
how many of these do you envision being in the bundle? are we talking MB's of output? If lots of output it might be better to replace with a summary plus the average stddev and the worst `min(25,10%)` followed by a list of all the job names